src/librustc_typeck/check/_match.rs

```diff
@@ -867,6 +868,15 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
             self.check_pat_walk(&field.pat, field_ty, def_bm, true);
         }
 
+        if tcx.sess.features.borrow().non_exhaustive {
```

Comment:
Unnecessary feature check.

Reply:
Fixed.

Reply:
After doing this, I noticed a regression in creating the structs - wasn't able to work out what was causing it.